core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
I think `JenkinsRule` doesn't allow you to set-up a different context than http://<JENKINS_URL>/jenkins

I took also a look at similar approaches in the test folder, but didn't really find any of them with a similar approach in which you really need to be in a different context. 

Do you think this should be something which need to be created there?